### ISSUE TITLE
Unexpected results from accumulated gradient

### ISSUE THREAD

**Issue by Crysitna** [pytorch/xla#958]
I was testing the accumulated gradient on xla to hedge the inadequate memory for large batch size. 

without loss accumulation:
```
for i, (input, target) in loader:
    pred = model(input)
    loss = loss_fn(pred, target)
    loss.backward()
    xm.optimizer_step(optimizer)
    model.zero_grad()
```

with loss accumulation:
```
total = 0
for i, (input, target) in loader:
    pred = model(input)
    loss = loss_fn(pred, target)
    loss = loss/2
    loss.backward()

    count= len(query)
    assert count== 16
    total += count
    if total % 32 == 0:
        xm.optimizer_step(optimizer)
        model.zero_grad()
```
yet encounter the following issue: 

without accumulated gradient: 
`batch-size = 32`: acc = 0.41

with accumulated gradient: 
`batch-size = 16 accumulate 2 batch` acc = 0.39

wondering if the gradient is additively accumulated in xla?

**Comment by artyompal** [comment]
Batch normalization could have caused this
